src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -30,9 +32,30 @@
 (def ServicesStatus
   {schema/Str ServiceStatus})
 
+(def SemVerVersion
+  (schema/pred semver/valid-format? "semver"))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
+(schema/defn ^:always-validate
+  get-service-version :- SemVerVersion
+  [group-id artifact-id]
+  (let [version (versioneer/get-version group-id artifact-id)]
+    (when-not version
+      (throw (IllegalStateException.
+               (format "Unable to find version number for '%s/%s'"
+                       group-id
+                       artifact-id))))
+    (when-not (semver/valid-format? version)
```

Comment:
Honestly, because I don't see any good reason to ever *not* mark a schemafied function as `:always-validate` until the day we see that causing a performance problem of some kind :)

I can remove it if you all would prefer, I just generally think the cost of those validations is so insignificant compared to anything interesting that our code does, it's never seemed worth spending any energy trying to decide when to use that and when not to.